pkg/backplane/ca/ca.go

```diff
@@ -0,0 +1,178 @@
+package ca
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"math/big"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	rsaBits  = 2048
+	validFor = 48 * time.Hour
+	caName   = "sonobuoy-ca"
+)
+
+var pkixName = pkix.Name{
+	Organization:       []string{"Heptio"},
+	OrganizationalUnit: []string{"sonobuoy"},
+	Country:            []string{"USA"},
+	Locality:           []string{"Seattle"},
+}
+
+// Authority represents a root certificate authority that can issues
+// certificates to be used for Client certs.
+// Sonobuoy issues every worker a client certificate
+type Authority struct {
+	privKey    *rsa.PrivateKey
+	cert       *x509.Certificate
+	lastSerial *big.Int
+}
+
+var randReader = rand.Reader
+
+// NewAuthority creates a new certificate authority. A new private key and root certificate will
+// be generated but not returned.
+func NewAuthority() (*Authority, error) {
+
+	privKey, err := rsa.GenerateKey(randReader, rsaBits)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't generate private key")
+	}
+	auth := &Authority{
+		privKey: privKey,
+	}
+	cert, err := auth.makeCert(privKey.Public(), func(cert *x509.Certificate) {
+		cert.IsCA = true
+		cert.KeyUsage = x509.KeyUsageCertSign
+		cert.Subject.CommonName = caName
+	})
+	if err != nil {
+		return nil, err
+	}
+	auth.cert = cert
+	return auth, nil
+}
+
+func (a *Authority) makeCert(pub crypto.PublicKey, mut func(*x509.Certificate)) (*x509.Certificate, error) {
```

Comment:
is `mut` short for mutate? I'd suggest either a comment or a more descriptive name (former if it's common lingo, latter if not)